src/test/addrman_tests.cpp

```diff
@@ -107,7 +54,7 @@ BOOST_FIXTURE_TEST_SUITE(addrman_tests, BasicTestingSetup)
 
 BOOST_AUTO_TEST_CASE(addrman_simple)
 {
-    auto addrman = std::make_unique<AddrManTest>();
+    auto addrman = std::make_unique<AddrMan>(std::vector<bool>(), /*deterministic=*/true, /*consistency_check_ratio=*/100);
```

Comment:
This line appears 17 times in the test. It's also very noisy and distracting from the purpose of the test. Perhaps add a utility function to create an addrman:

```diff
index a4d57e3435..28a754e079 100644
--- a/src/test/addrman_tests.cpp
+++ b/src/test/addrman_tests.cpp
@@ -49,12 +49,16 @@ static std::vector<bool> FromBytes(const unsigned char* source, int vector_size)
     return result;
 }
 
+static std::unique_ptr<AddrMan> TestAddrman(std::vector<bool> asmap = std::vector<bool>())
+{
+    return std::make_unique<AddrMan>(asmap, /*deterministic=*/true, /*consistency_check_ratio=*/100);
+}
 
 BOOST_FIXTURE_TEST_SUITE(addrman_tests, BasicTestingSetup)
 
 BOOST_AUTO_TEST_CASE(addrman_simple)
 {
-    auto addrman = std::make_unique<AddrMan>(std::vector<bool>(), /*deterministic=*/true, /*consistency_check_ratio=*/100);
+    auto addrman = TestAddrman();
 
     CNetAddr source = ResolveIP("252.2.2.2");
 
@@ -597,7 +601,7 @@ BOOST_AUTO_TEST_CASE(addrman_serialization)
 {
     std::vector<bool> asmap1 = FromBytes(asmap_raw, sizeof(asmap_raw) * 8);
 
-    auto addrman_asmap1 = std::make_unique<AddrMan>(asmap1, /*deterministic=*/true, /*consistency_check_ratio=*/100);
+    auto addrman_asmap1 = TestAddrman(asmap1);
     auto addrman_asmap1_dup = std::make_unique<AddrMan>(asmap1, /*deterministic=*/true, /*consistency_check_ratio=*/100);
     auto addrman_noasmap = std::make_unique<AddrMan>(std::vector<bool>(), /*deterministic=*/true, /*consistency_check_ratio=*/100);
     CDataStream stream(SER_NETWORK, PROTOCOL_VERSION);
```

(I only changed a couple of instances, but you get the idea)

Reply:
Done as suggested. These calls are also changed in #23373 but I think that these changes are compatible, i.e. `TestAddrman()` should call the `GetCheckRatio()` function introduced there.

Reply:
Looks good. Thanks!